File: Day 10 Training/inventory/src/main/java/com/example/WarehouseManager.java
package com.example;
import java.util.List;
import java.util.Scanner;

public class WarehouseManager {
    private Warehouse warehouse;
    private Scanner scanner;

    public WarehouseManager() {
        warehouse = new Warehouse();
        scanner = new Scanner(System.in);
    }

    public void start() {
        while (true) {
            showMenu();
            int choice = scanner.nextInt();
            scanner.nextLine(); // Consume newline

            switch (choice) {
                case 1:
                    addItem();
                    break;
                case 2:
                    updateItem();
                    break;
                case 3:
                    deleteItem();
                    break;
                case 4:
                    showItems();
                    break;
                case 5:
                    showItemQuantity();
                    break;
                case 6:
                    System.out.println("Exiting...");
                    return;
                default:
                    System.out.println("Invalid choice. Please try again.");
            }
        }
    }

    private void showMenu() {
        System.out.println("Warehouse Management System");
        System.out.println("1. Add Item");
        System.out.println("2. Update Item");
        System.out.println("3. Delete Item");
        System.out.println("4. Show All Items");
        System.out.println("5. Show Item Quantity");
        System.out.println("6. Exit");
        System.out.print("Enter your choice: ");
    }

    private void addItem() {
        System.out.print("Enter Item ID: ");
        String itemId = scanner.nextLine();
        System.out.print("Enter Item Title: ");
        String title = scanner.nextLine();
        System.out.print("Enter Item Price: ");
        double price = scanner.nextDouble();
        scanner.nextLine(); // Consume newline
        System.out.print("Enter Item Description: ");
        String description = scanner.nextLine();
        System.out.print("Enter Item Quantity: ");
        int quantity = scanner.nextInt();
        scanner.nextLine(); // Consume newline

        Item item = new Item(itemId, price, description, title);
        warehouse.addItem(item, quantity);
        System.out.println("Item added successfully!");
    }

    private void updateItem() {
        System.out.print("Enter Item ID to update: ");
        String itemId = scanner.nextLine();

        Item existingItem = warehouse.getItemById(itemId);
        if (existingItem == null) {
            System.out.println("Item not found!");
            return;
        }

        System.out.print("Enter new Item Title (leave blank to keep " + existingItem.getTitle() + "): ");
        String title = scanner.nextLine();
        if (title.isEmpty()) {
            title = existingItem.getTitle();
        }

        System.out.print("Enter new Item Price (leave blank to keep " + existingItem.getPrice() + "): ");
        String priceInput = scanner.nextLine();
        double price = priceInput.isEmpty() ? existingItem.getPrice() : Double.parseDouble(priceInput);

        System.out.print("Enter new Item Description (leave blank to keep " + existingItem.getDescription() + "): ");
        String description = scanner.nextLine();
        if (description.isEmpty()) {
            description = existingItem.getDescription();
        }

        System.out.print("Enter new Item Quantity (leave blank to keep " + warehouse.getItemQuantity(itemId) + "): ");
        String quantityInput = scanner.nextLine();
        int quantity = quantityInput.isEmpty() ? warehouse.getItemQuantity(itemId) : Integer.parseInt(quantityInput);

        boolean updated = warehouse.updateItem(itemId, price, description, title, quantity);
        if (updated) {
            System.out.println("Item updated successfully!");
        } else {
            System.out.println("Item not found!");
        }
    }

    private void deleteItem() {
        System.out.print("Enter Item ID to delete: ");
        String itemId = scanner.nextLine();

        boolean deleted = warehouse.deleteItem(itemId);
        if (deleted) {
            System.out.println("Item deleted successfully!");
        } else {
            System.out.println("Item not found!");
        }
    }

    private void showItems() {
        List<Item> items = warehouse.getItems();
        if (items.isEmpty()) {
            System.out.println("No items found.");
        } else {
            System.out.println("Items in the warehouse:");
            for (Item item : items) {
                System.out.println(item);
            }
        }
    }

    private void showItemQuantity() {
        System.out.print("Enter Item ID to check quantity: ");
        String itemId = scanner.nextLine();

        int quantity = warehouse.getItemQuantity(itemId);
        if (quantity == -1) {
            System.out.println("Item not found!");
        } else {
            System.out.println("Quantity of Item ID " + itemId + ": " + quantity);
        }
    }

    public static void main(String[] args) {
        WarehouseManager manager = new WarehouseManager();
        manager.start();
    }
}

File: Day 29 Training/NomanInvetoryMangement/src/main/kotlin/WarehouseManager.kt
package org.example

import java.util.Scanner

class WarehouseManager {
    private val warehouse = Warehouse()
    private val scanner = Scanner(System.`in`)

    fun start() {
        while (true) {
            showMenu()
            when (scanner.nextInt()) {
                1 -> addItem()
                2 -> updateItem()
                3 -> deleteItem()
                4 -> showItems()
                5 -> showItemQuantity()
                6 -> {
                    println("Exiting...")
                    return
                }
                else -> println("Invalid choice. Please try again.")
            }
        }
    }

    private fun showMenu() {
        println("Warehouse Management System")
        println("1. Add Item")
        println("2. Update Item")
        println("3. Delete Item")
        println("4. Show All Items")
        println("5. Show Item Quantity")
        println("6. Exit")
        print("Enter your choice: ")
    }

    private fun addItem() {
        println("Enter Item ID: ")
        val itemId = scanner.next()
        scanner.nextLine() // Clear the newline character from the buffer
        println("Enter Item Title: ")
        val title = scanner.nextLine()
        println("Enter Item Price: ")
        val price = scanner.nextDouble()
        scanner.nextLine() // Clear the newline character from the buffer
        println("Enter Item Description: ")
        val description = scanner.nextLine()
        println("Enter Item Quantity: ")
        val quantity = scanner.nextInt()

        val item = Item(itemId, price, description, title, quantity)
        if (warehouse.addItem(item)) {
            println("Item added successfully!")
        } else {
            println("Failed to add item.")
        }
    }

    private fun updateItem() {
        println("Enter Item ID to update: ")
        val itemId = scanner.next()

        val item = warehouse.getItemById(itemId)
        if (item == null) {
            println("Item not found!")
            return
        }

        scanner.nextLine() // Clear the newline character from the buffer
        println("Enter new Item Title (current: ${item.title}): ")
        val titleInput = scanner.nextLine()
        val title = if (titleInput.isBlank()) item.title else titleInput

        println("Enter new Item Price (current: ${item.price}): ")
        val priceInput = scanner.nextLine()
        val price = if (priceInput.isBlank()) item.price else priceInput.toDouble()

        println("Enter new Item Description (current: ${item.description}): ")
        val descriptionInput = scanner.nextLine()
        val description = if (descriptionInput.isBlank()) item.description else descriptionInput

        println("Enter new Item Quantity (current: ${item.quantity}): ")
        val quantityInput = scanner.nextLine()
        val quantity = if (quantityInput.isBlank()) item.quantity else quantityInput.toInt()

        if (warehouse.updateItem(itemId, price, description, title, quantity)) {
            println("Item updated successfully!")
        } else {
            println("Failed to update item.")
        }
    }

    private fun deleteItem() {
        println("Enter Item ID to delete: ")
        val itemId = scanner.next()

        if (warehouse.deleteItem(itemId)) {
            println("Item deleted successfully!")
        } else {
            println("Failed to delete item.")
        }
    }

    private fun showItems() {
        val items = warehouse.getItems()
        if (items.isEmpty()) {
            println("No items found.")
        } else {
            println("Items in the warehouse:")
            items.forEach { println(it) }
        }
    }

    private fun showItemQuantity() {
        println("Enter Item ID to check quantity: ")
        val itemId = scanner.next()

        val item = warehouse.getItemById(itemId)
        if (item == null) {
            println("Item not found!")
        } else {
            println("Quantity of Item ID $itemId: ${item.quantity}")
        }
    }
}

fun main() {
    WarehouseManager().start()
}



